app/views/ingredients/index.html.erb

```diff
@@ -17,7 +17,7 @@
             <td><%= ingredient.unit_type %></td>
             <td>
               <%= link_to 'Edit', edit_ingredient_path(ingredient), class: 'btn btn-primary', type: 'edit' %>
-              <%= link_to 'Delete', '#', class: 'btn btn-danger', type: 'delete' %>
+              <%= link_to 'Delete', { action: :destroy, id: ingredient.id }, class: 'btn btn-danger', method: :delete, data: { confirm: 'Are you sure?' } %>
```

Comment:
I think that you can replace `{ action: :destroy, id: ingredient.id }` with simple `ingredient_path(ingredient.id)`.